### PR TITLE
GR-1388 adapt migrate

### PR DESCRIPTION
**Pull request by 0xdcarns**
## Describe your changes

Utilized registration logic to handle migrations. Allows decoupling of access keys.
Ingress/Egress/Relay statuses are not kept.

## Provide Issue ticket number if applicable/not in title
title.

## Provide testing steps

- [x] attempt to install a new netclient over an existing 0.17.1 client.
- [x] Ensure it joins and has connectivity.

## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [x] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [x] My unit tests pass locally.
- [x] Netmaker is awesome.

### CODE REVIEW

File: controllers/migrate.go [hunk 1]
```diff
@@ -26,78 +24,63 @@ import (
...
-	if legacyNode.IsIngressGateway == "yes" {
-		if _, err := logic.CreateIngressGateway(legacyNode.Network, legacyNode.ID, false); err != nil {
-			logger.Log(0, "error creating ingress gateway during migration", err.Error())
+	if !logic.HostExists(&data.NewHost) {
+		if err = logic.CreateHost(&data.NewHost); err != nil {
+			logic.ReturnErrorResponse(w, r, logic.FormatError(err, "badrequest"))
+			return
 		}
 	}
-	if legacyNode.IsEgressGateway == "yes" {
```
Comment (mattkasun) [inline controllers/migrate.go:89]: why are egress gateways not recreated, all the information is available.

File: controllers/migrate.go [hunk 2]
```diff
@@ -26,78 +24,63 @@ import (
...
-	r.Body = io.NopCloser(strings.NewReader(string(payload)))
-	r.ContentLength = int64(len(string(payload)))
-	logger.Log(3, "deleteing legacy node", data.LegacyNodeID, legacyNode.ID, legacyNode.Name)
-	if err := database.DeleteRecord(database.NODES_TABLE_NAME, data.LegacyNodeID); err != nil {
-		logger.Log(0, "error deleting legacy node", legacyNode.Name, err.Error())
-	}
-	createNode(w, r)
-	//newly created node has same node id as legacy node allowing using legacyNode.ID in gateway creation
-	logger.Log(3, "re-creating legacy gateways")
-	if legacyNode.IsIngressGateway == "yes" {
```
Comment (mattkasun) [inline controllers/migrate.go:84]: why are ingress not recreated, all the info required for recreation is available

File: controllers/migrate.go [hunk 3]
```diff
@@ -26,78 +24,63 @@ import (
...
-		if _, err := logic.CreateEgressGateway(legacyNode.EgressGatewayRequest); err != nil {
-			logger.Log(0, "error creating egress gateway during migration", err.Error())
-		}
+	key, keyErr := logic.RetrievePublicTrafficKey()
+	if keyErr != nil {
+		logger.Log(0, "error retrieving key:", keyErr.Error())
+		logic.ReturnErrorResponse(w, r, logic.FormatError(err, "internal"))
+		return
 	}
-	if legacyNode.IsRelay == "yes" {
```
Comment (mattkasun) [inline controllers/migrate.go:94]: why not recreate relays.....

**Review by mattkasun — CHANGES_REQUESTED** [review]

File: controllers/migrate.go (hunk 3 above)
Comment (0xdcarns) [reply controllers/migrate.go:94]: the nodes are not created till host responds, see registration logic

**Review by afeiszli — APPROVED** [review]

**Comment by theguy951357** [comment]
testing notes in ticket 1388

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**